### PR TITLE
[WOR-816] Handle GoogleJsonResponseException, which appears to be wrapping the exception we expected.

### PR DESCRIPTION
**Pull request by cahrens**
Another iteration on https://broadworkbench.atlassian.net/browse/WOR-816, see comments on the review itself more Sentry links and more context.

---

**PR checklist**

- [ ] Include the JIRA issue number in the PR description and title
- [ ] Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] If you changed anything in `model/`, then you should [publish a new official `rawls-model`](https://github.com/broadinstitute/rawls/blob/develop/README.md#publish-rawls-model) and update `rawls-model` in [Orchestration's dependencies](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/project/Dependencies.scala).
- [ ] Get two thumbsworth of PR review
- [ ] Verify all tests go green, including CI tests
- [ ] **Squash commits and merge** to develop (branches are automatically deleted after merging)
- [ ] Inform other teams of any substantial changes via Slack and/or email

### CODE REVIEW

File: core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala [hunk 1]
```diff
@@ -2755,9 +2763,13 @@ class WorkspaceService(protected val ctx: RawlsRequestContext,
...
                                 expectedGoogleProjectPermissions
         )
-        .recoverWith { case t: IOException =>
-          // Throw a 400 to avoid Sentry notifications.
-          Future.failed(new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, t)))
+        .recoverWith {
+          case t: GoogleJsonResponseException =>
+            val code = getStatusCodeHandlingUnknown(t.getStatusCode)
+            Future.failed(new RawlsExceptionWithErrorReport(ErrorReport(code, t.getDetails.toString)))
+          case t: IOException =>
```
Comment (cahrens) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala:2770]: `GoogleJsonResponseException` extends `IOException`, so handle the more specific case first (we can get the status code for that!).

File: core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala [hunk 2]
```diff
@@ -2681,6 +2682,9 @@ class WorkspaceService(protected val ctx: RawlsRequestContext,
     getPermissionsFromRoles(googleRoles)
   }
 
+  private def getStatusCodeHandlingUnknown(intCode: Integer) =
```
Comment (cahrens) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala:2685]: I decided that while I'm mucking some more with this error handling, might as well fix this issue too.

https://broad-institute.sentry.io/issues/4044258270/?project=153297&query=is%3Aunresolved&referrer=issue-stream&stream_index=12

File: core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala [hunk 3]
```diff
@@ -2740,10 +2744,14 @@ class WorkspaceService(protected val ctx: RawlsRequestContext,
...
           expectedGoogleBucketPermissions
         )
-        .recoverWith { case t: StorageException =>
+        .recoverWith {
           // Throw with the status code of the exception (for example 403 for invalid billing, 400 for requester pays)
           // instead of a 500 to avoid Sentry notifications.
-          Future.failed(new RawlsExceptionWithErrorReport(ErrorReport(t.getCode, t)))
+          case t: StorageException =>
+            Future.failed(new RawlsExceptionWithErrorReport(ErrorReport(getStatusCodeHandlingUnknown(t.getCode), t)))
+          case t: GoogleJsonResponseException =>
```
Comment (cahrens) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala:2752]: It's looking like I should always expect the `StorageException` to be wrapped in a `GoogleJsonResponseException`, but I figure handling both isn't a big deal.

The 2 different stack traces showing here are pretty confusing to me-- it wasn't clear to me that StorageException wasn't the thing to handle: https://broad-institute.sentry.io/issues/3915484154/?project=153297&query=is%3Aignored&referrer=issue-stream&stream_index=1

File: core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceServiceSpec.scala [hunk 4]
```diff
@@ -3349,7 +3352,33 @@ class WorkspaceServiceSpec
...
+      )
+    )
+    val err = intercept[RawlsExceptionWithErrorReport] {
+      Await.result(services.workspaceService.checkWorkspaceCloudPermissions(testData.workspace.toWorkspaceName),
+                   Duration.Inf
+      )
+    }
+
+    err.errorReport.message should include(mockErrorMessage)
+    err.errorReport.statusCode.get.intValue() shouldBe 498
```
Comment (cahrens) [inline core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceServiceSpec.scala:3378]: 451-499 are not defined in StatusCodes.

File: core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala (hunk 3 above)
Comment (aherbst-broad) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala:2752]: That is really confusing, agreed it doesn't hurt to catch both.

**Review by aherbst-broad — APPROVED** [review]

**Review by marctalbott — APPROVED** [review]

**Pull request merged**